SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -1150,25 +1153,103 @@ void CNEMOEulerSolver::Source_Residual(CGeometry *geometry, CSolver **solver_con
+
+        /*--- Set solution  ---*/
+        numerics->SetConservative(nodes->GetSolution(iPoint), nodes->GetSolution(iPoint));
+
+        /*--- Set control volume ---*/
+        numerics->SetVolume(geometry->nodes->GetVolume(iPoint));
+
+        /*--- Set y coordinate ---*/
+        numerics->SetCoord(geometry->nodes->GetCoord(iPoint), geometry->nodes->GetCoord(iPoint));
+
+        /*--- Set primitive variables for viscous terms and/or generalised source ---*/
+        numerics->SetPrimitive(nodes->GetPrimitive(iPoint), nodes->GetPrimitive(iPoint));
+
+        /*--- If necessary, set variables needed for viscous computation ---*/       
+        if (viscous) {
+
+          /*--- Set gradient of primitive variables ---*/
+          numerics->SetPrimVarGradient(nodes->GetGradient_Primitive(iPoint), nodes->GetGradient_Primitive(iPoint));
+
+          /*--- Set gradient of auxillary variables ---*/
+          numerics->SetAuxVarGrad(nodes->GetAuxVarGradient(iPoint), nullptr);
+
+          /*--- Set diffusion coefficient ---*/
+          numerics->SetDiffusionCoeff(nodes->GetDiffusionCoeff(iPoint), nodes->GetDiffusionCoeff(iPoint));
+
+          /*--- Laminar viscosity ---*/
+          numerics->SetLaminarViscosity(nodes->GetLaminarViscosity(iPoint), nodes->GetLaminarViscosity(iPoint));
+
+          /*--- Eddy viscosity ---*/
+          numerics->SetEddyViscosity(nodes->GetEddyViscosity(iPoint), nodes->GetEddyViscosity(iPoint));
+
+          /*--- Thermal conductivity ---*/
+          numerics->SetThermalConductivity(nodes->GetThermalConductivity(iPoint), nodes->GetThermalConductivity(iPoint));
+
+          /*--- Vib-el. thermal conductivity ---*/
+          numerics->SetThermalConductivity_ve(nodes->GetThermalConductivity_ve(iPoint), nodes->GetThermalConductivity_ve(iPoint));
+
+          /*--- Vib-el energy ---*/          
+          numerics->SetEve(nodes->GetEve(iPoint), nodes->GetEve(iPoint));
+
+          /*--- Set turbulence kinetic energy ---*/
+          if (rans){
+            CVariable* turbNodes = solver_container[TURB_SOL]->GetNodes();
+            numerics->SetTurbKineticEnergy(turbNodes->GetSolution(iPoint,0), turbNodes->GetSolution(iPoint,0));
+          }
+        }
+
+        auto residual = numerics->ComputeAxisymmetric(config);
+
+        /*--- Check for errors before applying source to the linear system ---*/
+        err = false;
+        for (iVar = 0; iVar < nVar; iVar++)
+          if (residual[iVar] != residual[iVar]) err = true;
+        if (implicit)
+          for (iVar = 0; iVar < nVar; iVar++)
+            for (jVar = 0; jVar < nVar; jVar++)
+              if (Jacobian_i[iVar][jVar] != Jacobian_i[iVar][jVar]) err = true;
+
+        /*--- Apply the update to the linear system ---*/
+        if (!err) {
+          LinSysRes.AddBlock(iPoint, residual);
+          if (implicit)
+            Jacobian.AddBlock(iPoint, iPoint, Jacobian_i);
+        }else
+          eAxi_local++;
+      }
+
     }
```

Comment:
extra line between between the brackets

Reply:
Thanks for the review Catarina, I think I've addressed this and the other comments on formatting.